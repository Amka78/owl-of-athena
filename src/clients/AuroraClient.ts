import { Auth, Login, CreateUser, User } from "../types";
import RestClient from "./RestClient";

/**
 * Aurora Rest API
 *
 * @export
 * @class AuroraClient
 * @extends {RestClient}
 */
export class AuroraClient extends RestClient {
    constructor(baseUrl: string, locale: string) {
        super(baseUrl, {
            headers: {
                "Accept-Language": locale
            }
        });
    }

    /**
     * Signup
     *
     * @param {AuthDto} createUser
     * @returns {(Promise<TokenDto>)}
     * @memberof AuthClient
     */
    public async signup(createUser: CreateUser): Promise<User> {
        const response = await this.post<CreateUser>("users", createUser);

        if (response.ok) {
            return await response.json();
        }
        throw await response.json();
    }

    /**
     * Login
     *
     * @param {Login} login
     * @returns {(Promise<Auth>)}
     * @memberof AuroraClient
     */
    public async login(login: Login): Promise<Auth> {
        const response = await this.post<Login>("auth/email", login);

        if (response.ok) {
            return await response.json();
        }
        throw await response.json();
    }
}
