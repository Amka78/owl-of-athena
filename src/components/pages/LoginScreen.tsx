//#region Import Modules
import React, { FunctionComponent } from "react";

import { useLogin, useWindowDimensions } from "../../hooks";
import { LoginScreenTemplate } from "../templates/LoginScreenTemplate";
//#endregion

//#region Component
export const LoginScreen: FunctionComponent = () => {
    const loginHook = useLogin();
    const dimens = useWindowDimensions();

    return (
        <LoginScreenTemplate
            email={{
                ...loginHook.email,
            }}
            password={{
                ...loginHook.password,
            }}
            errorText={{
                children: loginHook.generalError,
            }}
            loginButton={{
                onPress: loginHook.onLoginPress,
            }}
            cancelButton={{
                onPress: loginHook.onCancelPress,
            }}
            forgotPasswordButton={{
                onPress: loginHook.onForgotPasswordPress,
            }}
            signupButton={{
                onPress: loginHook.onSignupPress,
            }}
            dimens={dimens}
        ></LoginScreenTemplate>
    );
};
//#endregion
