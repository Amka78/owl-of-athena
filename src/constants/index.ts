import ActionTypes from "./ActionTypes";
import Colors from "./Colors";
import Dimens from "./Dimens";
import Fonts from "./Fonts";
import Layout from "./Layout";
import Message, { MessageLocalizationParam } from "./Message";
import { MessageKeys } from "./MessageKeys";
import StorageKeys from "./StorageKeys";

export {
    ActionTypes,
    Colors,
    Dimens,
    Fonts,
    Layout,
    MessageLocalizationParam,
    Message,
    MessageKeys,
    StorageKeys
};
