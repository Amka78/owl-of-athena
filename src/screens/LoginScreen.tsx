import React, { FunctionComponent } from "react";
import { View } from "react-native";
import { CommonStyles } from "../styles";
import {
    Button,
    ContentTitle,
    ErrorText,
    FlatButton,
    TextBox
} from "../components";
import { useTextBox, useLogin } from "../hooks";
import { useNavigation } from "react-navigation-hooks";
export const LoginScreen: FunctionComponent = () => {
    const emailHooks = useTextBox("");
    const passwordHooks = useTextBox("");
    const { navigate } = useNavigation();
    const login = useLogin(false, {
        email: emailHooks.value,
        password: passwordHooks.value
    });
    return (
        <View style={CommonStyles.rootContainer}>
            <View style={CommonStyles.standardView}>
                <ContentTitle>{"login_title"}</ContentTitle>
                <View style={{ flex: 1 }}>
                    <TextBox
                        {...emailHooks}
                        label={"login_input_email"}
                        keyboardType={"email-address"}
                    ></TextBox>
                </View>
                <View style={{ flex: 1 }}>
                    <TextBox
                        {...passwordHooks}
                        label={"login_input_password"}
                        secureTextEntry={true}
                    ></TextBox>
                </View>
                <ErrorText>{login.generalError}</ErrorText>
                <Button {...login}>{"login_button"}</Button>
                <FlatButton
                    onPress={() => {
                        navigate("ForgotPassword");
                    }}
                >
                    {"login_forgot_password_button"}
                </FlatButton>
                <FlatButton
                    onPress={() => {
                        navigate("Signup");
                    }}
                >
                    {"login_no_account_button"}
                </FlatButton>
            </View>
        </View>
    );
};
