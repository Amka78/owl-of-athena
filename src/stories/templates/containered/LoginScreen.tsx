//#region Import Moduels
import React, { FunctionComponent } from "react";
import {
    LoginScreenTemplate,
    LoginScreenTemplateProps,
} from "../../../components/templates/LoginScreenTemplate";
import { useWindowDimensions } from "../../../hooks";
import { Container } from "../../Container";
//#endregion

//#region Types
export type LoginScreenProps = Omit<LoginScreenTemplateProps, "dimens">;
//#endregion

//#region Component
export const LoginScreen: FunctionComponent<LoginScreenProps> = (
    props: LoginScreenProps
) => {
    const dimens = useWindowDimensions();
    return (
        <Container dimens={dimens}>
            <LoginScreenTemplate
                {...props}
                dimens={dimens}
            ></LoginScreenTemplate>
        </Container>
    );
};
//#endregion
