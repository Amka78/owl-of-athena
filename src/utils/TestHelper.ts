import "../components";

import { shallow, ShallowWrapper } from "enzyme";
import enzymetoJson from "enzyme-to-json";
import {
    GestureResponderEvent,
    NativeSyntheticEvent,
    TextInputEndEditingEventData,
} from "react-native";

import { ButtonProps } from "../components/atoms/Button";

// @ts-ignore
export const FakeGestureResponderEvent: GestureResponderEvent = {};
// @ts-ignore
export const FakeTextBoxOnEndEditingEventData: NativeSyntheticEvent<TextInputEndEditingEventData> = {};

export function createMock<P, S>(comp: any): ShallowWrapper<P, S> {
    return shallow<P, S>(comp);
}

export function toJson(wrapper: any) {
    return enzymetoJson(wrapper);
}

/**
 * TestIDを利用してComponetのPropsを取得します。
 *
 * @static
 * @template T Propsの型を指定
 * @param {ShallowWrapper} component
 * @param {string} name
 * @param {string} testId
 * @returns {T}
 * @memberof TestHelper
 */
export function getComponentPropsByTestID<T>(
    component: ShallowWrapper<any, any>,
    name: string,
    testId: string
): T | undefined {
    const findComponent = component.find(name).findWhere((wrapper: any) => {
        const targetProps = wrapper.props();
        return targetProps.testID === testId;
    });
    if (findComponent.length <= 0) {
        return undefined;
    }
    const props: T = findComponent.props();
    return props;
}

export function getButtonPropsByTestID(
    component: ShallowWrapper<any, any>,
    testID: string
): ButtonProps | undefined {
    return getComponentPropsByTestID<ButtonProps>(
        component,
        "MultiLingualButton",
        testID
    );
}

/*public static getTextBoxPropsByTestID(
        component: ShallowWrapper<any, any>,
        testID: string
    ): ITextBoxProps | undefined {
        return this.getComponentPropsByTestID<ITextBoxProps>(
            component,
            "MultiLingualTextBox",
            testID
        );
    }

    public static getHeaderSearchBarPropsByTestID(
        component: ShallowWrapper<any, any>,
        testID: string
    ): IHeaderSearchBarProps | undefined {
        return this.getComponentPropsByTestID<IHeaderSearchBarProps>(
            component,
            "HeaderSearchBar",
            testID
        );
    }

    public static getListItemPropsByTestID(
        component: ShallowWrapper<any, any>,
        testID: string
    ): IListItemProps | undefined {
        return this.getComponentPropsByTestID<IListItemProps>(
            component,
            "MultiLingualListItem",
            testID
        );
    }*/

/*public static createNavigationMock(
        param: Partial<NavigationScreenProp<any, any>>
    ): NavigationScreenProp<any, any> {
        const navigation = new Object();
        Object.assign(navigation, param);
        return navigation as NavigationScreenProp<any, any>;
    }*/
