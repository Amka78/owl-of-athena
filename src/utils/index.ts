import BaseUrl from "./BaseUrl";
import PermissionUtil from "./PermissionUtil";
import TokenManager from "./TokenManager";
export { BaseUrl, PermissionUtil, TokenManager };
