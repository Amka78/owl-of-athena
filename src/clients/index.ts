import { AuroraClient } from "./AuroraClient";

export { AuroraClient };
