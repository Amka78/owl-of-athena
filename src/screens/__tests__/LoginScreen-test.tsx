import "jest-enzyme";
import "react-native";

import { ObjectID } from "bson";
import { ShallowWrapper } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import { Alert } from "react-native";

import { AuthClient } from "../../clients";
import { Message } from "../../constants";
import { AuthDto, TokenDto, UserDto } from "../../model";
import { TokenManager } from "../../utils";
import { FakeGestureResponderEvent, TestHelper } from "../../utils/TestHelper";
import LoginScreen, { IAuthProps, IState } from "../LoginScreen";

let component: ShallowWrapper<IAuthProps, IState, any>;
const mockAuthClient: AuthClient = new AuthClient("test", "test");
describe("LoginScreen-test", () => {
    beforeAll(async () => {
        Message.setLocale("ja-JP");
    });

    describe("when login", () => {
        beforeAll(async () => {
            const props = {
                authClient: mockAuthClient,
                isSignup: false,
                loginSuccess: undefined,
                navigation: undefined,
                token: "test"
            };
            // TODO
            // @ts-ignore
            component = TestHelper.createMock(<LoginScreen {...props} />);
        });

        beforeEach(async () => {
            component.setState({
                email: "",
                isEmailValid: false,
                isPasswordValid: false,
                password: "",
                passwordConfirm: "",
                showLoading: false
            });
            component = component.update();
        });
        it("The login form is displayed correctly", async () => {
            expect(component.length).toBe(1);

            const state = component.state();
            expect(state.email).toBe("");
            expect(state.isEmailValid).toBeFalsy();
            expect(state.isPasswordValid).toBeFalsy();
            expect(state.password).toBe("");
            expect(state.passwordConfirm).toBe("");
            expect(state.showLoading).toBeFalsy();

            // Two text boxes are displayed.
            expect(component.find("MultiLingualTextBox").length).toBe(2);
            // Login Button are displayed.
            expect(component.find("MultiLingualButton").length).toBe(1);

            expect(
                TestHelper.getButtonPropsByTestID(component, "loginButton")!
                    .localizedTitle
            ).toBe("login");

            expect(TestHelper.toJson(component)).toMatchSnapshot();
        });

        it.each(["", "test", "t@mail", ".com"])(
            "An error occurs if the mail is incorrect.",
            async (email: string) => {
                component.setState({ email });

                TestHelper.getTextBoxPropsByTestID(
                    component,
                    "accountIdTextBox"
                )!.onValidation!();

                component = component.update();

                expect(component.state().isEmailValid).toBeFalsy();
            }
        );

        it.each(["test@mail.com"])(
            "If there is no error in the mail address, no message is displayed.",
            async (email: string) => {
                component.setState({ email });

                TestHelper.getTextBoxPropsByTestID(
                    component,
                    "accountIdTextBox"
                )!.onValidation!();

                component = component.update();

                expect(component.state().isEmailValid).toBeTruthy();
            }
        );

        it.each(["", "t", "1234567"])(
            "Do not perform password check at login.",
            async (password: string) => {
                component.setState({ password });

                TestHelper.getTextBoxPropsByTestID(
                    component,
                    "passwordTextBox"
                )!.onValidation!();

                component = component.update();

                expect(component.state().isPasswordValid).toBeTruthy();
            }
        );

        it("Can log in when the correct value is entered.", async () => {
            expect(
                TestHelper.getButtonPropsByTestID(component, "loginButton")!
                    .disabled
            ).toBeTruthy();

            component.setState({
                isEmailValid: true
            });

            component = component.update();

            expect(
                TestHelper.getButtonPropsByTestID(component, "loginButton")!
                    .disabled
            ).toBeFalsy();
        });

        it("Can not log in unless the correct value is entered.", async () => {
            expect(
                TestHelper.getButtonPropsByTestID(component, "loginButton")!
                    .disabled
            ).toBeTruthy();

            component.setState({
                isEmailValid: false
            });

            component = component.update();

            expect(
                TestHelper.getButtonPropsByTestID(component, "loginButton")!
                    .disabled
            ).toBeTruthy();
        });

        it("submitLoginCredentials correct work", async () => {
            const testingEmail = "testuser@mail.com";
            const testingPassword = "test";
            component.setState({
                email: testingEmail,
                password: testingPassword
            });

            const navigate = jest.fn();
            const loginSuccess = jest.fn();

            component.setProps({
                loginSuccess,
                navigation: TestHelper.createNavigationMock({ navigate })
            });

            component.update();

            const testingUserId = new ObjectID("5c6daf4fe6e54f46319e4430");
            const testingToken = new TokenDto({
                token: "test",
                user: new UserDto({
                    id: testingUserId
                })
            });
            const loginMock = jest.fn(
                // @ts-ignore
                async (loginDto: AuthDto) => testingToken
            );

            const setMock = jest.fn();
            jest.spyOn(TokenManager, "set").mockImplementation(setMock);
            jest.spyOn(mockAuthClient, "login").mockImplementation(loginMock);

            await TestHelper.getButtonPropsByTestID(component, "loginButton")!
                .onPress!(FakeGestureResponderEvent);

            component = component.update();

            expect(component.state().showLoading).toBeFalsy();

            expect(loginMock.mock.calls.length).toBe(1);
            expect(loginMock.mock.calls[0][0].accountId).toBe(testingEmail);
            expect(loginMock.mock.calls[0][0].password).toBe(testingPassword);
            expect(loginSuccess.mock.calls.length).toBe(1);
            expect(loginSuccess.mock.calls[0][0].token).toBe(
                testingToken.token
            );
            expect(setMock.mock.calls.length).toBe(1);
            expect(setMock.mock.calls[0][0].token).toBe(testingToken.token);
            expect(setMock.mock.calls[0][0].user).toBe(
                testingUserId.toString()
            );
            expect(navigate.mock.calls.length).toBe(1);
            expect(navigate.mock.calls[0][0]).toBe("UserForm");
        });

        it("An error occurs if saving fails.", async () => {
            component.setState({
                email: "testuser@mail.com",
                password: "test"
            });

            const alertMock = jest.fn();

            jest.spyOn(mockAuthClient, "login").mockImplementation(() => {
                throw { error: "testError", message: "testMessage" };
            });
            jest.spyOn(Alert, "alert").mockImplementation(alertMock);

            await TestHelper.getButtonPropsByTestID(component, "loginButton")!
                .onPress!(FakeGestureResponderEvent);

            expect(alertMock.mock.calls.length).toBe(1);
            expect(alertMock.mock.calls[0][0]).toBe("testError");
            expect(alertMock.mock.calls[0][1]).toBe("testMessage");
        });
    });

    describe("when signup", () => {
        beforeAll(async () => {
            const props = {
                authClient: mockAuthClient,
                isSignup: true,
                loginSuccess: undefined,
                navigation: undefined,
                token: "test"
            };
            // TODO
            // @ts-ignore
            component = TestHelper.createMock(<LoginScreen {...props} />);
        });

        beforeEach(async () => {
            component.setState({
                email: "",
                isEmailValid: false,
                isPasswordValid: false,
                password: "",
                passwordConfirm: "",
                showLoading: false
            });
            component = component.update();
        });

        it("The signup form is displayed correctly", async () => {
            // Two text boxes are displayed.
            expect(component.find("MultiLingualTextBox").length).toBe(3);
            // Signup Button are displayed.
            expect(component.find("MultiLingualButton").length).toBe(1);

            expect(
                TestHelper.getButtonPropsByTestID(component, "loginButton")!
                    .localizedTitle
            ).toBe("create_account");

            expect(toJson(component)).toMatchSnapshot();
        });

        it.each(["", "t", "1234567"])(
            "An error occurs if the password is incorrect.",
            async (password: string) => {
                component.setState({ password });
                TestHelper.getTextBoxPropsByTestID(
                    component,
                    "passwordTextBox"
                )!.onValidation!();

                component = component.update();

                expect(component.state().isPasswordValid).toBeFalsy();
            }
        );

        it.each(["12345678", "123456789"])(
            "If there is no error in the password, no message is displayed.",
            async (password: string) => {
                component.setState({ password });
                TestHelper.getTextBoxPropsByTestID(
                    component,
                    "passwordTextBox"
                )!.onValidation!();

                component = component.update();

                expect(component.state().isPasswordValid).toBeTruthy();
            }
        );

        it.each(["12345678"])(
            "An error occurs if the password and password comfirm does not mouch.",
            async (password: string) => {
                component.setState({
                    password,
                    passwordConfirm: password + 1
                });

                TestHelper.getTextBoxPropsByTestID(
                    component,
                    "passwordConfirmTextBox"
                )!.onValidation!();

                component = component.update();

                expect(component.state().isPasswordConfirmValid).toBeFalsy();
            }
        );

        it.each(["12345678"])(
            "If there is no error in the passwordConfrim, no message is displayed.",
            async (password: string) => {
                component.setState({
                    password,
                    passwordConfirm: password
                });

                TestHelper.getTextBoxPropsByTestID(
                    component,
                    "passwordConfirmTextBox"
                )!.onValidation!();

                component = component.update();

                expect(component.state().isPasswordConfirmValid).toBeTruthy();
            }
        );

        it.each([
            [false, false, false],
            [true, false, false],
            [false, true, false],
            [true, true, false],
            [false, false, true],
            [true, false, true],
            [false, true, true]
        ])(
            "Can not sign up unless the correct value is entered.",
            async (
                isEmailValid: boolean,
                isPasswordValid: boolean,
                isPasswordConfirmValid: boolean
            ) => {
                expect(
                    TestHelper.getButtonPropsByTestID(component, "loginButton")!
                        .disabled
                ).toBeTruthy();

                component.setState({
                    isEmailValid,
                    isPasswordConfirmValid,
                    isPasswordValid
                });

                component = component.update();

                expect(
                    TestHelper.getButtonPropsByTestID(component, "loginButton")!
                        .disabled
                ).toBeTruthy();
            }
        );

        it("Can Signup when the correct value is entered.", async () => {
            const props = TestHelper.getButtonPropsByTestID(
                component,
                "loginButton"
            )!;
            expect(props.disabled).toBeTruthy();

            component.setState({
                isEmailValid: true,
                isPasswordConfirmValid: true,
                isPasswordValid: true
            });

            component = component.update();

            expect(
                TestHelper.getButtonPropsByTestID(component, "loginButton")!
                    .disabled
            ).toBeFalsy();
        });

        it("Succeed with signing up.", async () => {
            const testingEmail = "testuser@mail.com";
            const testingPassword = "test";
            component.setState({
                email: testingEmail,
                password: testingPassword,
                passwordConfirm: testingPassword
            });

            const navigate = jest.fn();
            const loginSuccess = jest.fn();

            component.setProps({
                loginSuccess,
                navigation: TestHelper.createNavigationMock({ navigate })
            });

            const testingUserId = new ObjectID("5c6daf4fe6e54f46319e4430");
            const testingToken = new TokenDto({
                token: "test",
                user: new UserDto({
                    id: testingUserId
                })
            });
            const signinMock = jest.fn(
                // @ts-ignore
                async (loginDto: AuthDto) => testingToken
            );

            const setMock = jest.fn();
            jest.spyOn(TokenManager, "set").mockImplementation(setMock);
            jest.spyOn(mockAuthClient, "signup").mockImplementation(signinMock);

            await TestHelper.getButtonPropsByTestID(component, "loginButton")!
                .onPress!(FakeGestureResponderEvent);

            component = component.update();

            expect(component.state().showLoading).toBeFalsy();

            expect(signinMock.mock.calls.length).toBe(1);
            expect(signinMock.mock.calls[0][0].accountId).toBe(testingEmail);
            expect(signinMock.mock.calls[0][0].password).toBe(testingPassword);
            expect(loginSuccess.mock.calls.length).toBe(1);
            expect(loginSuccess.mock.calls[0][0].token).toBe(
                testingToken.token
            );
            expect(setMock.mock.calls.length).toBe(1);
            expect(setMock.mock.calls[0][0].token).toBe(testingToken.token);
            expect(setMock.mock.calls[0][0].user).toBe(
                testingUserId.toString()
            );
            expect(navigate.mock.calls.length).toBe(1);
            expect(navigate.mock.calls[0][0]).toBe("UserForm");
        });
    });
});
