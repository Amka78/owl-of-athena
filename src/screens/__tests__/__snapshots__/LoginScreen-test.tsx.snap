// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen-test when login The login form is displayed correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <ImageBackground
    source={1}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": 1334,
        "justifyContent": "center",
        "left": 0,
        "top": 0,
        "width": 750,
      }
    }
  >
    <KeyboardAvoidingView
      behavior="padding"
      enabled={true}
      keyboardVerticalOffset={100}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "height": 400,
            "marginTop": 10,
            "width": 300,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontFamily": "mont-bold",
                  "fontSize": 40,
                }
              }
            >
              Aracan
            </Text>
          </View>
        </View>
        <MultiLingualTextBox
          autoFocus={false}
          blurOnSubmit={true}
          containerStyle={
            Object {
              "marginVertical": 5,
            }
          }
          errorMessage="正しいメールアドレスを入力してください。"
          inputStyle={
            Object {
              "color": "white",
              "marginLeft": 10,
            }
          }
          keyboardType="email-address"
          leftIcon={
            <Themed.Icon
              color="rgba(171, 189, 219, 1)"
              name="account-box"
              size={25}
              type="material-community"
            />
          }
          localizedPlaceholder="e_mail"
          onChangeText={[Function]}
          onValidation={[Function]}
          placeholderTextColor="white"
          returnKeyType="next"
          testID="accountIdTextBox"
          value=""
        />
        <MultiLingualTextBox
          blurOnSubmit={true}
          containerStyle={
            Object {
              "marginVertical": 10,
            }
          }
          inputStyle={
            Object {
              "color": "white",
              "marginLeft": 10,
            }
          }
          keyboardType="default"
          leftIcon={
            <Themed.Icon
              color="rgba(171, 189, 219, 1)"
              name="lock"
              size={25}
              type="material-community"
            />
          }
          localizedPlaceholder="password"
          onChangeText={[Function]}
          onValidation={[Function]}
          placeholderTextColor="white"
          returnKeyType="done"
          secureTextEntry={true}
          testID="passwordTextBox"
          value=""
        />
        <MultiLingualButton
          buttonStyle={
            Object {
              "borderRadius": 30,
              "height": 40,
              "width": 200,
            }
          }
          containerStyle={
            Object {
              "alignItems": "center",
              "marginVertical": 20,
            }
          }
          disabled={true}
          loading={false}
          loadingProps={
            Object {
              "color": "white",
              "size": "small",
            }
          }
          localizedTitle="login"
          onPress={[Function]}
          testID="loginButton"
          titleStyle={
            Object {
              "fontWeight": "bold",
            }
          }
        />
      </View>
    </KeyboardAvoidingView>
  </ImageBackground>
</View>
`;

exports[`LoginScreen-test when signup The signup form is displayed correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <ImageBackground
    source={1}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": 1334,
        "justifyContent": "center",
        "left": 0,
        "top": 0,
        "width": 750,
      }
    }
  >
    <KeyboardAvoidingView
      behavior="padding"
      enabled={true}
      keyboardVerticalOffset={100}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "height": 400,
            "marginTop": 10,
            "width": 300,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontFamily": "mont-bold",
                  "fontSize": 40,
                }
              }
            >
              Aracan
            </Text>
          </View>
        </View>
        <MultiLingualTextBox
          autoFocus={false}
          blurOnSubmit={true}
          containerStyle={
            Object {
              "marginVertical": 5,
            }
          }
          errorMessage="正しいメールアドレスを入力してください。"
          inputStyle={
            Object {
              "color": "white",
              "marginLeft": 10,
            }
          }
          keyboardType="email-address"
          leftIcon={
            <Themed.Icon
              color="rgba(171, 189, 219, 1)"
              name="account-box"
              size={25}
              type="material-community"
            />
          }
          localizedPlaceholder="e_mail"
          onChangeText={[Function]}
          onValidation={[Function]}
          placeholderTextColor="white"
          returnKeyType="next"
          testID="accountIdTextBox"
          value=""
        />
        <MultiLingualTextBox
          blurOnSubmit={true}
          containerStyle={
            Object {
              "marginVertical": 10,
            }
          }
          errorMessage="パスワードは8桁以上を入力してください。"
          inputStyle={
            Object {
              "color": "white",
              "marginLeft": 10,
            }
          }
          keyboardType="default"
          leftIcon={
            <Themed.Icon
              color="rgba(171, 189, 219, 1)"
              name="lock"
              size={25}
              type="material-community"
            />
          }
          localizedPlaceholder="password"
          onChangeText={[Function]}
          onValidation={[Function]}
          placeholderTextColor="white"
          returnKeyType="done"
          secureTextEntry={true}
          testID="passwordTextBox"
          value=""
        />
        <MultiLingualTextBox
          blurOnSubmit={true}
          containerStyle={
            Object {
              "marginVertical": 10,
            }
          }
          errorMessage="パスワードと確認用パスワードが一致しません。"
          inputStyle={
            Object {
              "color": "white",
              "marginLeft": 10,
            }
          }
          keyboardType="default"
          leftIcon={
            <Themed.Icon
              color="rgba(171, 189, 219, 1)"
              name="lock"
              size={25}
              type="material-community"
            />
          }
          localizedPlaceholder="password_confirm"
          onChangeText={[Function]}
          onValidation={[Function]}
          placeholderTextColor="white"
          returnKeyType="done"
          secureTextEntry={true}
          testID="passwordConfirmTextBox"
          value=""
        />
        <MultiLingualButton
          buttonStyle={
            Object {
              "borderRadius": 30,
              "height": 40,
              "width": 200,
            }
          }
          containerStyle={
            Object {
              "alignItems": "center",
              "marginVertical": 20,
            }
          }
          disabled={true}
          loading={false}
          loadingProps={
            Object {
              "color": "white",
              "size": "small",
            }
          }
          localizedTitle="create_account"
          onPress={[Function]}
          testID="loginButton"
          titleStyle={
            Object {
              "fontWeight": "bold",
            }
          }
        />
      </View>
    </KeyboardAvoidingView>
  </ImageBackground>
</View>
`;
